src/routing/ProtectedRoute.jsx

```diff
@@ -11,34 +11,29 @@ import { getDecodedParams } from "utils/decoding"
 axios.defaults.withCredentials = true
 
 const ProtectedRoute = ({ children, component: WrappedComponent, ...rest }) => {
-  return (
-    <LoginConsumer>
-      {({ userId }) =>
-        userId ? (
-          children ||
-          (WrappedComponent && (
-            <Route
+  const { userId } = useLoginContext()
+  return userId ? (
+    children ||
+      (WrappedComponent && (
+        <Route
+          // eslint-disable-next-line react/jsx-props-no-spreading
+          {...rest}
+          render={(props) => {
+            const { match } = props
+            const { params } = match
+            const newMatch = {
+              ...match,
+              decodedParams: getDecodedParams(params),
+            }
+            return (
               // eslint-disable-next-line react/jsx-props-no-spreading
-              {...rest}
-              render={(props) => {
-                const { match } = props
-                const { params } = match
-                const newMatch = {
-                  ...match,
-                  decodedParams: getDecodedParams(params),
-                }
-                return (
-                  // eslint-disable-next-line react/jsx-props-no-spreading
-                  <WrappedComponent {...rest} {...props} match={newMatch} />
-                )
-              }}
-            />
-          ))
-        ) : (
-          <Redirect to="/" />
-        )
-      }
-    </LoginConsumer>
+              <WrappedComponent {...rest} {...props} match={newMatch} />
+            )
+          }}
+        />
+      ))
+  ) : (
+    <Redirect to="/" />
```

Comment:
skimmed; assuming only change is to use the hook call rather than the render props pattern